packages/react-core/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Icon/icon';
+import { css } from '@patternfly/react-styles';
+import { Spinner } from '../Spinner';
+
+export interface IconComponentProps extends Omit<React.HTMLProps<HTMLSpanElement>, 'size'> {
+  /** Icon content when isInProgress is set to false */
+  children?: React.ReactNode;
+  /** Icon content when isInProgress is set to true. Defaults to a medium spinner with 1em diameter.  */
+  progressIcon?: React.ReactNode;
+  /** Additional classes applied to the icon container */
+  className?: string;
+  /** Size of the icon component container and icon. */
+  size?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Size of icon content. Overrides the icon size set by the size property. */
+  iconSize?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Status color of the icon */
+  status?: 'default' | 'info' | 'success' | 'warning' | 'danger';
+  /** Indicates the icon is inline and should inherit the text font size and color. Overriden by size and iconSize properties. */
+  isInline?: boolean;
+  /** Indicates the icon is in progress. Setting this property to true will swap the icon with the progressIcon. */
+  isInProgress?: boolean;
+  /** Aria-label for the default progress icon */
+  defaultProgressArialabel?: string;
+}
+
+export const Icon: React.FunctionComponent<IconComponentProps> = ({
+  children,
+  className,
+  progressIcon,
+  size,
+  iconSize,
+  status,
+  isInline = false,
+  isInProgress = false,
+  defaultProgressArialabel = 'Loading...',
+  ...props
+}: IconComponentProps) => {
+  const _progressIcon = progressIcon ? (
+    progressIcon
+  ) : (
+    <Spinner size="md" diameter="1em" isSVG aria-label={defaultProgressArialabel} />
+  );
+
+  return (
+    <span
+      className={css(
+        styles.icon,
+        isInline && styles.modifiers.inline,
+        isInProgress && styles.modifiers.inProgress,
+        styles.modifiers[size],
+        className
+      )}
+      {...props}
+    >
+      <span className={css(styles.iconContent, styles.modifiers[iconSize], styles.modifiers[status])}>{children}</span>
```

Comment:
@mcoker this differs slightly from the HTML implementation.  The "progress" icon would only render when `isProgress` is true.  Would that cause an issue with dynamically changing icons?

Reply:
@tlabaj it shouldn't, at least as far as the way the component works now. The non-progress icon is what creates the space/size in the icon component layout. The progress icon is absolutely positioned in the icon layout, meaning it doesn't take up any space or have a size as far as the icon layout is concerned.

The only reason we may want the progress icon in the layout before it's rendered is if we
* animate the progress icon to fade it in - it would need to exist (if only shortly, using a super quick `setTimeout()` could do the trick) before it's "shown" so that it can fade in as opposed to just appearing
* want the progress icon to take up space and be part of the icon component layout when it isn't shown

Neither of those are true, so I think it's fine as it is.